src/libcore/ops.rs

```diff
@@ -64,6 +64,20 @@
 //!
 //! See the documentation for each trait for a minimum implementation that
 //! prints something to the screen.
+//!
+//! This example shows the behavior of the various `Range*` structs.
+//!
+//! ```rust
+//! let arr = [0, 1, 2, 3, 4];
+//!
+//! assert_eq!(arr[ .. ], [0,1,2,3,4]); // RangeFull
+//! assert_eq!(arr[ ..3], [0,1,2    ]); // RangeTo
+//! assert_eq!(arr[1.. ], [  1,2,3,4]); // RangeFrom
+//! assert_eq!(arr[1..3], [  1,2    ]); // Range
+//!
+//! assert_eq!(arr[ ...3], [0,1,2,3 ]); // RangeToIncusive
+//! assert_eq!(arr[1...3], [  1,2,3 ]); // RangeInclusive
```

Comment:
We could just add a note to that effect.

On Aug 17, 2016 2:59 PM, "Steve Klabnik" <notifications@github.com> wrote:

> In src/libcore/ops.rs
> <https://github.com/rust-lang/rust/pull/35759#discussion_r75185100>:
>
> > @@ -64,6 +64,20 @@
> >  //!
> >  //! See the documentation for each trait for a minimum implementation that
> >  //! prints something to the screen.
> > +//!
> > +//! This example shows the behavior of the various `Range*` structs.
> > +//!
> > +//! ```rust
> > +//! let arr = [0, 1, 2, 3, 4];
> > +//!
> > +//! assert_eq!(arr[ .. ], [0,1,2,3,4]); // RangeFull
> > +//! assert_eq!(arr[ ..3], [0,1,2    ]); // RangeTo
> > +//! assert_eq!(arr[1.. ], [  1,2,3,4]); // RangeFrom
> > +//! assert_eq!(arr[1..3], [  1,2    ]); // Range
> > +//!
> > +//! assert_eq!(arr[ ...3], [0,1,2,3 ]); // RangeToIncusive
> > +//! assert_eq!(arr[1...3], [  1,2,3 ]); // RangeInclusive
>
> I know you copied these from the existing examples, but I wonder if we
> should eliminate the extra spacing, since it's not idiomatic Rust. Hmm
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-lang/rust/pull/35759/files/7564da22d890784af83bcdebac404c2d6d674bf7#r75185100>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGfGaWOEno8eN_VLR0fTdvOGkxKdZBxTks5qg1oVgaJpZM4JmxaP>
> .
>